force-app/main/default/classes/selectors/tests/FundingRequestSelector_TEST.cls

```diff
@@ -0,0 +1,204 @@
+                    .isAccessible()
+            );
+
+            System.assertEquals(
+                false,
+                Schema.Funding_Request__c.SObjectType.fields.Total_Disbursed__c.getDescribe()
+                    .isAccessible()
+            );
+
+            System.assertEquals(
+                false,
+                Schema.Funding_Request__c.SObjectType.fields.Total_Remaining__c.getDescribe()
+                    .isAccessible()
+            );
+
+            System.assertEquals(
+                false,
+                Schema.Funding_Request__c.SObjectType.fields.Total_Remaining__c.getDescribe()
+                    .isAccessible()
+            );
+            // Question regarding lookups
+
+            // Run the test
+            String fundingRequestId = [SELECT Id FROM Funding_Request__c LIMIT 1].Id;
+
+            final FundingRequestSelector fundingRequestSelector = new FundingRequestSelector();
+
+            try {
+                List<Funding_Request__c> fundingRequests = fundingRequestSelector.getFundingRequestById(
+                    fundingRequestId
+                );
+            } catch (System.QueryException e) {
+                actualException = e;
+            }
+        }
+
+        Test.stopTest();
+
+        System.assertNotEquals(
+            null,
+            actualException,
+            'A System.QueryException should have been thrown because the user cannot read Funding_Request__c.'
+        );
+    }
+
+    @isTest
+    private static void callFundingRequest_WithCrudAccess() {
+        Test.startTest();
+
+        System.assert(Schema.Funding_Request__c.SObjectType.getDescribe().isAccessible());
+
+        System.assert(Schema.Contact.SObjectType.getDescribe().isAccessible());
+
+        System.assert(
+            Schema.Funding_Request__c.SObjectType.fields.Requested_Amount__c.getDescribe()
+                .isAccessible()
+        );
+
+        System.assert(
+            Schema.Funding_Request__c.SObjectType.fields.Awarded_Amount__c.getDescribe()
+                .isAccessible()
+        );
+
+        System.assert(
+            Schema.Funding_Request__c.SObjectType.fields.Total_Disbursed__c.getDescribe()
+                .isAccessible()
+        );
+
+        System.assert(
+            Schema.Funding_Request__c.SObjectType.fields.Total_Remaining__c.getDescribe()
+                .isAccessible()
+        );
+
+        System.assert(
+            Schema.Funding_Request__c.SObjectType.fields.Total_Remaining__c.getDescribe()
+                .isAccessible()
+        );
+
+        // Run the test
+        String fundingRequestId = [SELECT Id FROM Funding_Request__c LIMIT 1].Id;
+
+        final FundingRequestSelector fundingRequestSelector = new FundingRequestSelector();
+
+        List<Funding_Request__c> fundingRequests = fundingRequestSelector.getFundingRequestById(
+            fundingRequestId
+        );
+
+        Test.stopTest();
+
+        System.assertNotEquals(
```

Comment:
Do you mean to compare a Null with a Boolean here? Could we `assertNotEquals` `True` with `isEmpty()` ?

Reply:
Good catch... I'm updating it now